packages/deployment-server-ui/src/pages/Types.tsx

```diff
@@ -1,86 +1,143 @@
+
+      const deployRes = await fetch("/api/deployments");
+      const deploys = await deployRes.json();
+      setDeployments(
+        deploys.filter(
+          // @ts-ignore: ignore type warning
+          (deployment) =>
+            deployment.deviceId === findDeviceById(deployment.deviceId)?.id
+        )
+      );
+    };
+
+    fetchAsync();
   }, []);
 
+  useEffect(() => {
+    if (
+      devices != null &&
+      deviceType != null &&
+      deployments != null &&
+      loading
+    ) {
+      setLoading(false);
+      console.log(deviceType);
+    }
+  }, [devices, deviceType, deployments]);
+
+  useEffect(() => {
+    const fetchAsync = async () => {
+      const typesRes = await fetch(`/api/device-types/${id}`);
+      const types = await typesRes.json();
+      setDeviceType(types);
+
+      const deployRes = await fetch("/api/deployments");
+      const deploys = await deployRes.json();
+      setDeployments(
+        deploys.filter(
+          // @ts-ignore: Unreachable code error
+          (deployment) =>
+            deployment.deviceId === findDeviceById(deployment.deviceId)?.id
+        )
+      );
+    };
+
+    fetchAsync();
+  }, [refetchFlip]);
+
   return (
     <>
-      <h2>Types</h2>
-      <Divider />
-      <List
-        style={{
-          overflow: "scroll",
-          overflowX: "hidden",
-          padding: "24px",
-          maxHeight: "calc(100% - 60px)",
-        }}
-        grid={{ gutter: 16 }}
-        dataSource={types}
-        renderItem={(type, index) => {
-          if (isType(type)) {
-            return (
-              <List.Item key={type.id}>
-                <Card
-                  hoverable
-                  cover={
-                    <img
-                      alt={`${type.name} example image`}
-                      height={220}
-                      src="https://picsum.photos/300/220"
-                    />
-                  }
-                  actions={[<FilterOutlined />, <InfoOutlined />]}
-                >
-                  <Meta
-                    title={type.name}
-                    description={type.fqbn}
-                    style={{ padding: "24px 0px" }}
-                  />
-                </Card>
-              </List.Item>
-            );
-          } else {
-            return (
-              <List.Item key={`placeholder-${index}`}>
-                <Card
-                  hoverable
-                  cover={
-                    <Skeleton.Image
-                      style={{ width: "300px", height: "220px" }}
+      <Row gutter={[{ xs: 8, sm: 16, md: 24, lg: 32 }, 20]} align="middle">
+        <Col span={24}>
+          <List
+            header={<h2>Supported Board Types</h2>}
+            style={{
+              overflow: "scroll",
+              overflowX: "hidden",
+              padding: "24px",
+              maxHeight: "750px",
+            }}
+            grid={{ gutter: 32 }}
+            dataSource={typesData.data}
```

Comment:
Hardcoding the data like this causes huge issues due to the unique ids. You already fetch the required data. Please map this onto the correct svgs and data from other fetched sources.